lib/widgets/button.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+
+enum ViamButtonRole {
+  primary,
+  inverse,
+  success,
+  danger,
+  warning;
+
+  Color get backgroundColor {
+    final brightness = SchedulerBinding.instance.platformDispatcher.platformBrightness;
+    final isDarkMode = brightness == Brightness.dark;
+    switch (this) {
+      case ViamButtonRole.primary:
+        return isDarkMode ? const Color.fromARGB(255, 40, 40, 41) : const Color.fromARGB(255, 240, 240, 240);
+      case ViamButtonRole.inverse:
+        return isDarkMode ? const Color.fromARGB(255, 240, 240, 240) : const Color.fromARGB(255, 40, 40, 41);
+      case ViamButtonRole.success:
+        return isDarkMode ? const Color.fromARGB(255, 105, 153, 103) : const Color.fromARGB(255, 61, 125, 63);
+      case ViamButtonRole.danger:
+        return isDarkMode ? const Color.fromARGB(255, 211, 103, 94) : const Color.fromARGB(255, 190, 53, 54);
+      case ViamButtonRole.warning:
+        return isDarkMode ? const Color.fromARGB(255, 250, 185, 82) : const Color.fromARGB(255, 242, 166, 0);
+    }
+  }
+
+  Color get foregroundColor {
+    final brightness = SchedulerBinding.instance.platformDispatcher.platformBrightness;
+    final isDarkMode = brightness == Brightness.dark;
+    switch (this) {
+      case ViamButtonRole.primary:
+        return isDarkMode ? const Color.fromARGB(255, 240, 240, 240) : const Color.fromARGB(255, 40, 40, 41);
+      case ViamButtonRole.inverse:
+        return isDarkMode ? const Color.fromARGB(255, 40, 40, 41) : const Color.fromARGB(255, 240, 240, 240);
+      case ViamButtonRole.success:
+        return const Color.fromARGB(255, 255, 255, 255);
+      case ViamButtonRole.danger:
+        return const Color.fromARGB(255, 255, 255, 255);
+      case ViamButtonRole.warning:
+        return const Color.fromARGB(255, 255, 255, 255);
+    }
+  }
+
+  ButtonStyle get style =>
+      ButtonStyle(backgroundColor: MaterialStatePropertyAll(backgroundColor), foregroundColor: MaterialStatePropertyAll(foregroundColor));
+}
+
+enum ViamButtonStyle {
+  filled,
+  outline,
+  ghost;
+}
+
+enum ViamButtonVariant {
+  iconOnly,
+  iconLeading,
+  iconTrailing;
+}
+
+class ViamButton extends StatelessWidget {
```

Comment:
what is the use of a viamButton? does this mean rather than someone using an ElevatedButton from flutter and styling it to match the viam theme, then they could just use a viamButton? Or is there greater functionality?

Reply:
It's basically just that -- just a button that's always going to be styled for Viam (regardless of what your theme is). I mostly made this for use internally for any/all buttons in the SDK widgets, and figured we might as well publish it like we do for PRIME.